src/hooks/__tests__/useChannel.test.tsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import PubNub from 'pubnub';
+import { cleanup, render } from '@testing-library/react';
+import { PubNubProvider } from '../../context/PubNubContext';
+import { useChannel } from '../useChannel';
+
+describe('useChannel hook', () => {
+  afterEach(cleanup);
+
+  let pubNubClient: PubNub;
+
+  beforeEach(() => {
+    if (pubNubClient) {
+      pubNubClient.unsubscribeAll();
+    }
+
+    pubNubClient = new PubNub({
+      publishKey: 'x',
+      subscribeKey: 'y',
+    });
+  });
+
+  it('should return publish and unsubscribe functions', () => {
+    const Child = () => {
+      const { publish, unsubscribe } = useChannel(['myChannel'], message => {
+        console.log(message);
+      });
+
+      expect(publish).toBeDefined();
+      expect(unsubscribe).toBeDefined();
+
+      return null;
+    };
+
+    render(
+      <PubNubProvider client={pubNubClient}>
+        <Child />
+      </PubNubProvider>
+    );
+  });
+
+  it('should call subscribe and addListeners as soon as mounted', () => {
+    const Child = () => {
+      const channel = useChannel(['myChannel'], message => {
```

Comment:
Neat! Though this seems more like an `useAutoSubscribe` or something vs `useChannel`. Also this might be overkill if someone uses this within a chat application where each sent message invokes another subscribe.

@MessagingMoore Thoughts?